qiskit/algorithms/time_evolvers/time_evolution_result.py

```diff
@@ -27,20 +28,27 @@ class TimeEvolutionResult(AlgorithmResult):
         aux_ops_evaluated (ListOrDict[tuple[complex, complex]] | None): Optional list of
             observables for which expected values on an evolved state are calculated. These values
             are in fact tuples formatted as (mean, standard deviation).
+        observables (ListOrDict[tuple[np.ndarray, np.ndarray]] | None): Optional list of
```

Comment:
Right now the evaluation times are just an element in the `observables`. That's a bit weird if the observables are passed as list, since there is no dictionary key specifying that the time is there. Could we instead make the `times` a separate attribute on the result object?